File: examples/javascript/neofollower.mjs
/**
 * Minimal NeoFollower Reseller API example.
 *
 * Requires Node.js 18+ for built-in fetch.
 *
 * Run:
 *   NEOFOLLOWER_API_KEY="YOUR_API_KEY" node neofollower.mjs
 */

const API_URL = "https://panel.neofollower.com/api/v1";
const API_KEY = process.env.NEOFOLLOWER_API_KEY;

if (!API_KEY) {
  throw new Error("Set NEOFOLLOWER_API_KEY first.");
}

async function requestApi(payload) {
  const body = new URLSearchParams({
    ...payload,
    key: API_KEY,
  });

  const response = await fetch(API_URL, {
    method: "POST",
    headers: {
      "Content-Type": "application/x-www-form-urlencoded",
      Accept: "application/json",
    },
    body,
  });

  if (!response.ok) {
    throw new Error(`NeoFollower API HTTP error: ${response.status}`);
  }

  return response.json();
}

export function services() {
  return requestApi({ action: "services" });
}

export function addOrder({ service, link, quantity, ...extra }) {
  return requestApi({
    action: "add",
    service: String(service),
    link,
    quantity: String(quantity),
    ...extra,
  });
}

export function status(orderId) {
  return requestApi({
    action: "status",
    order: String(orderId),
  });
}

export function multipleStatus(orderIds) {
  return requestApi({
    action: "status",
    orders: orderIds.join(","),
  });
}

export function balance() {
  return requestApi({ action: "balance" });
}

console.log(await balance());

File: examples/python/neofollower.py
"""Minimal NeoFollower Reseller API example.

Install:
    pip install requests

Run:
    export NEOFOLLOWER_API_KEY="YOUR_API_KEY"
    python neofollower.py
"""

import os
from typing import Any

import requests

API_URL = "https://panel.neofollower.com/api/v1"
API_KEY = os.environ["NEOFOLLOWER_API_KEY"]


def request_api(**payload: Any) -> Any:
    payload["key"] = API_KEY
    response = requests.post(API_URL, data=payload, timeout=30)
    response.raise_for_status()
    return response.json()


def services() -> Any:
    return request_api(action="services")


def add_order(service: int, link: str, quantity: int, **extra: Any) -> Any:
    return request_api(
        action="add",
        service=service,
        link=link,
        quantity=quantity,
        **extra,
    )


def status(order_id: int) -> Any:
    return request_api(action="status", order=order_id)


def multiple_status(order_ids: list[int]) -> Any:
    return request_api(
        action="status",
        orders=",".join(str(order_id) for order_id in order_ids),
    )


def balance() -> Any:
    return request_api(action="balance")


if __name__ == "__main__":
    print(balance())
